### PR TITLE
Insecure Register Receiver Flag

### PR DESCRIPTION
**Pull request by elyousfi5**

### CODE REVIEW

File: MOBILE_CLIENT/ANDROID/_MEDIUM/INSECURE_REGISTER_RECEIVER_FLAG/description.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+The application calls the registerReceiver method with the argument flags set to RECEIVER_EXPORTED, which can be exploitable as it exposes the BroadcastReceiver to external applications, potentially leading to unauthorized access and security vulnerabilities.
+
+=== "Java"
```
Comment (ErebusZ) [inline MOBILE_CLIENT/ANDROID/_MEDIUM/INSECURE_REGISTER_RECEIVER_FLAG/description.md:3]: Maybe add a vulnerable manifest file snippet.

Comment (elyousfi5) [reply MOBILE_CLIENT/ANDROID/_MEDIUM/INSECURE_REGISTER_RECEIVER_FLAG/description.md:3]: the KB is related only to the specific issue of passing flag RECEIVER_EXPORTED to registerReceiver. It doesnt have anything to do with manifest file.

File: MOBILE_CLIENT/ANDROID/_MEDIUM/INSECURE_REGISTER_RECEIVER_FLAG/description.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+The application calls the registerReceiver method with the argument flags set to RECEIVER_EXPORTED, which can be exploitable as it exposes the BroadcastReceiver to external applications, potentially leading to unauthorized access and security vulnerabilities.
```
Comment (BlueSquare1) [inline MOBILE_CLIENT/ANDROID/_MEDIUM/INSECURE_REGISTER_RECEIVER_FLAG/description.md:1]:
```suggestion
The application calls the registerReceiver method with the argument flags set to `RECEIVER_EXPORTED`, which can be exploitable as it exposes the BroadcastReceiver to external applications, potentially leading to unauthorized access and other security vulnerabilities.
```

File: MOBILE_CLIENT/ANDROID/_MEDIUM/INSECURE_REGISTER_RECEIVER_FLAG/meta.json [hunk 3]
```diff
@@ -0,0 +1,27 @@
+{
+  "risk_rating": "medium",
+  "short_description": "The application calls the registerReceiver method with the argument flags set to RECEIVER_EXPORTED.",
+  "references": {
+    "Shared Preferences (Android Developer)": "https://developer.android.google.cn/about/versions/13/features#runtime-receivers"
```
Comment (3asm) [inline MOBILE_CLIENT/ANDROID/_MEDIUM/INSECURE_REGISTER_RECEIVER_FLAG/meta.json:5]: Shared preference?

File: MOBILE_CLIENT/ANDROID/_MEDIUM/INSECURE_REGISTER_RECEIVER_FLAG/meta.json [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    ],
+    "GDPR": [
+      "ART_32"
+    ],
+    "PCI_STANDARDS":[
+      "REQ_2_2",
+      "REQ_6_2",
+      "REQ_6_3",
+      "REQ_11_3"
+    ]
```
Comment (3asm) [inline MOBILE_CLIENT/ANDROID/_MEDIUM/INSECURE_REGISTER_RECEIVER_FLAG/meta.json:25]: Are these relevant?

File: MOBILE_CLIENT/ANDROID/_MEDIUM/INSECURE_REGISTER_RECEIVER_FLAG/recommendation.md [hunk 5]
```diff
@@ -0,0 +1,11 @@
+To mitigate risks associated with exporting receivers in Android applications, export only when essential. Additionally, ensure the exported BroadcastReceiver is protected with the necessary permissions to minimize potential security vulnerabilities.
+
+=== "XML"
+	```xml
+    <receiver android:name=".MyReceiver" android:exported="true">
```
Comment (3asm) [inline MOBILE_CLIENT/ANDROID/_MEDIUM/INSECURE_REGISTER_RECEIVER_FLAG/recommendation.md:5]: No permission is set?

**Review by 3asm — CHANGES_REQUESTED** [review]

File: MOBILE_CLIENT/ANDROID/_MEDIUM/INSECURE_REGISTER_RECEIVER_FLAG/meta.json (hunk 3 above)
Comment (elyousfi5) [reply MOBILE_CLIENT/ANDROID/_MEDIUM/INSECURE_REGISTER_RECEIVER_FLAG/meta.json]: Fixed

File: MOBILE_CLIENT/ANDROID/_MEDIUM/INSECURE_REGISTER_RECEIVER_FLAG/meta.json (hunk 4 above)
Comment (elyousfi5) [reply MOBILE_CLIENT/ANDROID/_MEDIUM/INSECURE_REGISTER_RECEIVER_FLAG/meta.json:25]: Yes

**Pull request merged**